### PR TITLE
Long Factorials using Vector in cpp

### PR DESCRIPTION
**Pull request by Saviour1001**
### Have you read the [Contributing Guidelines on Pull Requests]
Yes

### Description

The algorithm of finding long factorials is already present in the algorithm database but isn't that understandable. It uses a lot of inbuilt functions and thus is hard to follow. I have added a different way to find the output and have explained it much better. Thus I have enhanced the existing algorithm.

### Checklist

- [ Yes] I've read the contribution guidelines.
- [ Yes] I've checked the issue list before deciding what to submit.
- [ Yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

This PR solves issue #2452
I had tried merging another PR earlier #2496 but it had conflicts with the Readme file. So I am closing #2496  and this PR will be the alternative.

### CODE REVIEW

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+/*Sample test cases 
+1. Input: 5
+   Output: 120
+
+2. Input: 15
+   Output: 1307674368000
+
+// Time Complexity = O(N)
+ */
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp]: Add a new line at the EOF

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+/*Sample test cases 
+1. Input: 5
+   Output: 120
+
+2. Input: 15
+   Output: 1307674368000
+
+// Time Complexity = O(N)
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp]: Comment inside a Comment?

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    factorial(num);
+    return 0;
+}
+
+/*Sample test cases 
+1. Input: 5
+   Output: 120
+
+2. Input: 15
+   Output: 1307674368000
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/long_factorials.cpp]: Keep only one sample I/O

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Format your Code properly

**Comment by Saviour1001** [comment]
Did the changes which were asked. @HarshCasper

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+the datatypes are not even capable of storing such a big number. We could achieve calculation of
+such a number via going back to the basis of how we used to muiltiply two numbers back in school days. 
+We used to muiltiply numbers and would add the summation and get the final result. 
+This program here uses a vector to store each particular digit in a single index of the vector. 
+The carry is forwarded as basic maths and in the end the complete result gets stored in the vector*/
+
+//The length of vector should be >= length of the resultant factorial
+
+
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/long_factorials.cpp]: Remove the extra two lines one is enough

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#include <vector>
+#include <iostream>
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/long_factorials.cpp]: Replace this header file with `#include<bits/stdc++.h>`

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+int main() {
+    int num;
+    cin>>num;
+    factorial(num);
+    return 0;
+}
+
+/*Sample test case 
+1. Input: 15
+   Output: 1307674368000
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/long_factorials.cpp]: Remove that 1. and space before I/P and O/P

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+Time Complexity = O(N)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/long_factorials.cpp:53]: Add Space Complexity at the end of the Code

File: C-Plus-Plus/cp/long_factorials.cpp [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+ */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/long_factorials.cpp:55]: Add a new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Saviour1001** [comment]
Did the changes @ankitaggarwal23

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make README changes as well

**Comment by Saviour1001** [comment]
Done @HarshCasper

**Review by HarshCasper — APPROVED** [review]
